### DISCUSSION TITLE
[Question] Project direction

### DISCUSSION THREAD

**Discussion by alexluong** [remirror/remirror#87] [Q&A]
Hey @ifiokjr,

Thanks for creating a very interesting library. Excited to see how it turns out.

My questions are:

- What's your objectives with `remirror`?
- Do you see it becoming the [`tiptap`](https://github.com/scrumpy/tiptap) of React? Or is this just an experiment?

Just wanna make sure if I should commit to the library. I intend to use it on a side project (not corporate), so I don't mind the experimental state of this. Just wanna be a bit certain that you see this being around in the near future.

Thanks!

**Comment by ifiokjr** [comment]
Hey @alexluong! I'm glad you stumbled across this library and are willing to use it while still in its infancy. 

Firstly, while the API's in this library are experimental, I don't see this library and its future as experimental. Working on `remirror` has captivated me. Iterating towards better API's while deepening my own understanding of React, TypeScript, the JS ecosystem, and the art of programming is an opportunity I'm deeply grateful for.

For a few months, things went quiet after I started with a new client. I found I had no time left to work on `remirror`. I've had to adapt to my approach and over the last few weeks, I feel like I've managed to find a better balance. Shout out to @jashmenn for reaching out and encouraging me to keep working on the library.

In short, I want to see this through to the end, and I'm hoping that I can create something meaningful that the open source community feels able to contribute to.

So to answer your questions, Tiptap transformed the way I viewed the Prosemirror API and how it could be reworked into something user-friendly and framework compatible. Even though `remirror` brings a very similar API for interacting with Prosemirror into the react world, my goals are different.

I want `remirror` to be a universal text editor that runs on the web, on the server and in React Native. I'm currently finishing off a PR which will provide automatic server-side support when rendering in a non-browser environment.

I've started hacking on a branch with React Native support and I think I have a solution (in my head) for making that integration possible.

I'm currently not using `remirror` for any projects so I'm very much led by users. SSR was prioritised because people mentioned it was what they needed the most. 

I'd be keen to know what kind of features you're looking for in your editor.

**Comment by alexluong** [comment]
Appreciate the detailed response, @ifiokjr.

It's awesome to hear that you want to see `remirror` through. I'd start using it in the upcoming weeks and come back later with feedback/bug reports/enhancement ideas.

As for the time being, I'm working on a WYSIWYG editor that ideally also supports most markdown syntax. What I means is that, for example, if I type `# Hello World`, I should see `Hello World` in `h1` block.

The end experience I strive for is similar to Notion's editor.

Also, super excited to hear that you aim to support React Native. I worked on Slate before and mobile/RN support was one of our issues (along with collaboration limitation). Anyway, looking forward to seeing where the library may go!

**Comment by hennessyevan** [comment]
Good to hear @ifiokjr let me know if you need any help. I’m thinking of adding an image extension maybe with filepond if that’d  be helpful

**Comment by ifiokjr** [comment]
> As for the time being, I'm working on a WYSIWYG editor that ideally also supports most markdown syntax. What I means is that, for example, if I type `# Hello World`, I should see `Hello World` in `h1` block.

@alexluong this is certainly possible and to an extent is already implemented with `inputRules`. I think bold and italic should just work already. Headers could also be added quite trivially.

This weekend and, quite probably the next few, my focus will be on a `react native` integration. This is the final foundational piece after which we can build more features with confidence. 

I'm happy to accept PR's though if you're keen and have time. I think it's something similar to the `BoldExtension`

https://github.com/ifiokjr/remirror/blob/c50cef31538c1b44de05c4d25e2d7947b6d28be4/%40remirror/core-extensions/src/marks/bold.ts#L48-L50

or `BlockquoteExtension`: 

https://github.com/ifiokjr/remirror/blob/c50cef31538c1b44de05c4d25e2d7947b6d28be4/%40remirror/core-extensions/src/nodes/blockquote.ts#L51-L53

I haven't thought it through so I'm not sure exactly the difficulties you'll face.

I'm also happy to arrange a call / hangout if you'd like me to walk you through some of the code (and this goes for anyone who wants to contribute). Send me a dm on [twitter](https://twitter.com/ifiokjr) and we can arrange something.

**Comment by ifiokjr** [comment]
> Good to hear @ifiokjr let me know if you need any help. I’m thinking of adding an image extension maybe with filepond if that’d be helpful

@hennessyevan I appreciate this a lot and I'd love your contributions! 

One thing to be aware of:  I'll be working on the `ReactNative` integration this weekend and I'm pretty sure certain things will arise that require an API change.

There's a risk that anything you build will need significant changes within the next few weeks. That being said, it may be a good idea to open an issue where we can discuss a feature set for any potential image extension.

**Comment by alexluong** [comment]
Hey I'd love to see the `RN` integration. Once you're done, I bet everyone would love to read some documentations regarding that and any changes in API.

> I'm happy to accept PR's though if you're keen and have time. I think it's something similar to the BoldExtension

I'd be able to make a PR after I get familiar with the codebase. I've built the input rule for it when I was still experimenting with `prosemirror` and `react`. It's very achievable!

**Comment by ifiokjr** [comment]
I'm closing this to clean up issues which don't require further actions.
